utils/docker/Dockerfile.el.8

```diff
@@ -93,6 +93,7 @@ RUN if $QUICKBUILD; then                                          \
 
 ARG BULLSEYE
 RUN if [ "x$BULLSEYE" != "x" ]; then \
+      dnf install zip; \
```

Comment:
This should be in utils/scripts/install-el8.sh obviously.

Reply:
Should it?  It's only needed when doing the Bullseye build.

Reply:
Yes, we have lots of packages there which are only needed for Java builds etc.

Reply:
But the Java build is not conditional and defaulted to off, is it?

I don't really feel all that strongly about this, but just trying to avoid unnecessary bloat (for (al)most (all)) everyone.

But really, if `zip` is being included to create the `.zip`'ed artifact, maybe it can be gotten rid of by addressing my other question about the usefulness of zipping up the HTML.

Reply:
Having all the packages in one place makes things simpler but yes it does mean potentially you're installed things that you don't need to, however the alternative is many install scripts with associated maintenance burden, plus the extra run-time and failure points when running the dockerfile.

That said the Java build does it's own downloads of anything it needs from the java side anyway.

+1 on the zip comments, we shouldn't really be zipping things up and if we do need to bundle them then tar + gz would be preferable if bullseye has that option.

Reply:
+1 on your +1 including the `.tar.gz` alternative to `.zip`.  But we really need to examine the usefulness of any kind of bundling of the HTML into single file.

Reply:
A few years ago, Jenkins put security fixes for what its web server is allowed to serve.  The HTML content generated for by Bullseye for the Clover plugin is blocked by those fixes and can not be displayed.

Zip archive format was used because the .tar.gz format is not easily used on Windows platforms, and also supported on the platforms.

The zipping of the web site for local download and display is  workaround until SRE can work on a better solution.

Reply:
Bullseye is a commercial product and requires a commercial license key to install/use.
The dockerfile section that uses BULLSEYE can not be used outside of our environment, as in third-parties using the Dockerfile, or by github actions.

CI had bullseye builds off by default because CI runs in the past had high failure rates on Unit testing with Bullseye.  Most seemed to be from the instrumentation slowing down some tests or otherwise affecting the run.   The Bullesye instrumented runs also added considerable more time to CI runs than they appear to be doing now.

CI Functional testing requires that the RPMs be built with Bullseye instrumentation built in, and those instrumented RPMs binaries do not appear to be in their own package with their own names, so enabling that changes both the RPM artifacts and all functional testing to be run with the instrumentation on.

Reply:
If you're going to add the dnf install step here then you need a "dnf clean all" command after it to remove any temporary files.

Reply:
> A few years ago, Jenkins put security fixes for what its web server is allowed to serve. The HTML content generated for by Bullseye for the Clover plugin is blocked by those fixes and can not be displayed.

Have we re-evaluated this recently to see if anything has changed for the positive in Jenkins?  It should be pretty trivial to archive what is usually put into the _zip_ to see what the rendering looks like currently.